lib/Doctrine/DBAL/Driver/ResultStatement.php

```diff
@@ -58,29 +58,53 @@ public function setFetchMode($fetchMode, $arg2 = null, $arg3 = null);
+     *                               \PDO::prepare().
+     * @param int $cursorOffset      For a ResultStatement object representing a scrollable cursor for which the
+     *                               cursorOrientation parameter is set to \PDO::FETCH_ORI_ABS, this value
+     *                               specifies the absolute number of the row in the result set that shall be
+     *                               fetched.
+     *                               For a ResultStatement object representing a scrollable cursor for which the
+     *                               cursorOrientation parameter is set to \PDO::FETCH_ORI_REL, this value
+     *                               specifies the row to fetch relative to the cursor position before
+     *                               ResultStatement::fetch() was called.
      *
      * @return mixed The return value of this method on success depends on the fetch mode. In all cases, FALSE is
      *               returned on failure.
      *
      * @see PDO::FETCH_* constants.
      */
-    public function fetch($fetchMode = null);
+    public function fetch($fetchMode = null, $cursorOrientation = \PDO::FETCH_ORI_NEXT, $cursorOffset = 0);
 
     /**
      * Returns an array containing all of the result set rows.
      *
-     * @param integer|null $fetchMode Controls how the next row will be returned to the caller.
-     *                                The value must be one of the PDO::FETCH_* constants,
-     *                                defaulting to PDO::FETCH_BOTH.
+     * @param int|null $fetchMode     Controls how the next row will be returned to the caller.
+     *                                The value must be one of the \PDO::FETCH_* constants,
+     *                                defaulting to \PDO::FETCH_BOTH.
+     * @param int|null $fetchArgument This argument has a different meaning depending on the value of the $fetchMode parameter:
+     *                                * \PDO::FETCH_COLUMN: Returns the indicated 0-indexed column.
+     *                                * \PDO::FETCH_CLASS: Returns instances of the specified class, mapping the columns of each
+     *                                  row to named properties in the class.
+     *                                * \PDO::FETCH_FUNC: Returns the results of calling the specified function, using each row's
+     *                                  columns as parameters in the call.
+     * @param array|null $ctorArgs    Controls how the next row will be returned to the caller.
+     *                                The value must be one of the \PDO::FETCH_* constants,
+     *                                defaulting to \PDO::FETCH_BOTH.
      *
      * @return array
      *
-     * @see PDO::FETCH_* constants.
+     * @see \PDO::FETCH_* constants.
      */
-    public function fetchAll($fetchMode = null);
+    public function fetchAll($fetchMode = null, $fetchArgument = null, $ctorArgs = null);
```

Comment:
This signature change is a BC break and needs upgrade/migration path in `UPGRADE.md`